src/DryIoc/Container.cs

```diff
@@ -6762,6 +6763,8 @@ public static FactoryMethodSelector Constructor(bool mostResolvable = false, boo
                 ctors = cs;
             else if (ctorsOrCtorOrType is Type t)
                 ctors = t.GetConstructors(BindingFlags.Public | BindingFlags.Instance | additionalToPublicAndInstance);
+            else if (reflectionFactory.ImplementationType is Type it)
```

Comment:
In this test, reflection factory instance is actually `WithTypeProvider`, and ImplementationType property has additional validation checks, that's why I decided to use it:

```csharp
public override Type ImplementationType
{
    get
    {
        var m = Made;
        var implType = ((Func<Type>)_implementationTypeOrProviderOrPubCtorOrCtors)();
        var validatedImplType = ValidateImplementationType(implType, m);
        if (IsFactoryGenerator(validatedImplType ?? implType, m))
            _generatedFactoriesOrFactoryGenerator = ImHashMap<KV<Type, object>, ReflectionFactory>.Empty;
        return validatedImplType;
    }
}
```

So my thoughts are:

1. Func<Type> typecast shouldn't be duplicated in Container code
2. ValidateImplementationType check is a good thing and it shouldn't be skipped
3. If reflection factory instance is something completely different and _implementationTypeOrProviderOrPubCtorOrCtors is neither Type, nor CtorInfo, nor Func<Type>, my approach will still work because overridden ImplementationType property will return a Type.

That's why I chose to use ImplementationType property.
But I can hardcode the Func<Type> branch if you still think it's a cleaner way.

Reply:
Ok, I hardcoded the match against Func<​Type> instead of using the ImplementationType property to avoid mixing different layers of abstraction :)